add doc for record OOM alarm (#11103)
pingcap/docs: system-variables.md

</CustomContent>

- Scope: GLOBAL
- Persists to cluster: Yes
- Type: Float
- Default value: `0.7`
- Range: `[0.0, 1.0]`

> - This variable sets the memory usage ratio that triggers the tidb-server memory alarm. By default, TiDB prints an alarm log when TiDB memory usage exceeds 70% of total memory and [alarm condition](/configure-memory-usage.md#tidb-server-alarm-on-excessive-memory-usage) is met.

Author: ```suggestion
- This variable sets the memory usage ratio that triggers the tidb-server memory alarm. By default, TiDB prints an alarm log when TiDB memory usage exceeds 70% of total memory and any [alarm condition](/configure-memory-usage.md#tidb-server-alarm-on-excessive-memory-usage) is met.
```

> - This variable sets the memory usage ratio that triggers the tidb-server memory alarm. By default, TiDB prints an alarm log when TiDB memory usage exceeds 70% of total memory and any [alarm condition](/configure-memory-usage.md#tidb-server-alarm-on-excessive-memory-usage) is met.

Author: ```suggestion
- This variable sets the memory usage ratio that triggers the tidb-server memory alarm. By default, TiDB prints an alarm log when TiDB memory usage exceeds 70% of its total memory and any of the [alarm conditions](/configure-memory-usage.md#tidb-server-alarm-on-excessive-memory-usage) is met.
```

> - When the value of this variable is configured to `0` or `1`, it means the memory threshold alarm feature is disabled.

Author: ```suggestion
- When this variable is configured to `0` or `1`, it means the memory threshold alarm feature is disabled.
```

<CustomContent platform="tidb">

> - When the value of this variable is configured to a value between `0` and `1`, it means that the memory threshold alarm feature is enabled.

Author: ```suggestion
- When this variable is configured to a value between `0` and `1`, it means that the memory threshold alarm feature is enabled.
```

    - If the [`server-memory-quota`](/tidb-configuration-file.md#server-memory-quota-introduced-from-v409-version) configuration is not set, the memory alarm threshold is `tidb_memory-usage-alarm-ratio * system memory size`.
    - If the `server-memory-quota` configuration is set and greater than 0, the memory alarm threshold is `tidb_memory-usage-alarm-ratio * server-memory-quota`.

</CustomContent>

<CustomContent platform="tidb-cloud">

> - When the value of this variable is configured to a value between `0` and `1`, it means that the memory threshold alarm feature is enabled.

Author: ```suggestion
- When this variable is configured to a value between `0` and `1`, it means that the memory threshold alarm feature is enabled.
```

    - If the [`server-memory-quota`](https://docs.pingcap.com/tidb/stable/tidb-configuration-file#memory-usage-alarm-ratio-new-in-v409) configuration is not set, the memory alarm threshold is `tidb_memory-usage-alarm-ratio * system memory size`.
    - If the `server-memory-quota` configuration is set and greater than 0, the memory alarm threshold is `tidb_memory-usage-alarm-ratio * server-memory-quota`.

</CustomContent>

### `tidb_memory_usage_alarm_keep_record_num` <span class="version-mark">New in v6.4.0</span>

- Scope: GLOBAL

> - Whether to persist to the cluster: Yes

Author: ```suggestion
- Persists to cluster: Yes
```

- Default value: `5`
- Range: `[1, 10000]`

> - When the tidb-server memory usage exceeds the memory alarm threshold and triggers an alarm, TiDB only retains the status file generated during the recent 5 alarms by default. You can adjust this number with this variable.

Author: ```suggestion
- When the tidb-server memory usage exceeds the memory alarm threshold and triggers an alarm, TiDB only retains the status files generated during the recent 5 alarms by default. You can adjust this number with this variable.
```

- When this variable is configured to a value greater than `0` and less than `1`, it means that the memory threshold alarm feature is enabled.

    - If the system variable [`tidb_server_memory_limit`](#tidb_server_memory_limit-new-in-v640) is `0` and the [`server-memory-quota`](/tidb-configuration-file.md#server-memory-quota-introduced-from-v409-version) configuration item is not set, the memory alarm threshold is `tidb_memory-usage-alarm-ratio * system memory size`.
    - If the system variable `tidb_server_memory_limit` is `0` and the `server-memory-quota` configuration item is set to greater than 0, the memory alarm threshold is `tidb_memory-usage-alarm-ratio * server-memory-quota`.
    - If the system variable `tidb_server_memory_limit` is set to greater than 0, the memory alarm threshold is `tidb_memory-usage-alarm-ratio * tidb_server_memory_limit`.

</CustomContent>

<CustomContent platform="tidb-cloud">

- When this variable is configured to a value greater than `0` and less than `1`, it means that the memory threshold alarm feature is enabled.

</CustomContent>

### `tidb_memory_usage_alarm_keep_record_num` <span class="version-mark">New in v6.4.0</span>

> 

Author: ```suggestion

<CustomContent platform="tidb-cloud">
> **Note:**
>
> This TiDB variable is not applicable to TiDB Cloud.
</CustomContent>

```

- When this variable is configured to a value greater than `0` and less than `1`, it means that the memory threshold alarm feature is enabled.

    - If the system variable [`tidb_server_memory_limit`](#tidb_server_memory_limit-new-in-v640) is `0` and the [`server-memory-quota`](/tidb-configuration-file.md#server-memory-quota-introduced-from-v409-version) configuration item is not set, the memory alarm threshold is `tidb_memory-usage-alarm-ratio * system memory size`.
    - If the system variable `tidb_server_memory_limit` is `0` and the `server-memory-quota` configuration item is set to greater than 0, the memory alarm threshold is `tidb_memory-usage-alarm-ratio * server-memory-quota`.
    - If the system variable `tidb_server_memory_limit` is set to greater than 0, the memory alarm threshold is `tidb_memory-usage-alarm-ratio * tidb_server_memory_limit`.

</CustomContent>

<CustomContent platform="tidb-cloud">

- When this variable is configured to a value greater than `0` and less than `1`, it means that the memory threshold alarm feature is enabled.

</CustomContent>

### `tidb_memory_usage_alarm_keep_record_num` <span class="version-mark">New in v6.4.0</span>

> <CustomContent platform="tidb-cloud">
> > **Note:**
> >
> > This TiDB variable is not applicable to TiDB Cloud.
> </CustomContent>

Author: ```suggestion
<CustomContent platform="tidb-cloud">

> **Note:**
>
> This TiDB variable is not applicable to TiDB Cloud.

</CustomContent>
```

</CustomContent>

- Scope: GLOBAL
- Persists to cluster: Yes
- Type: Float
- Default value: `0.7`
- Range: `[0.0, 1.0]`

> - This variable sets the memory usage ratio that triggers the tidb-server memory alarm. By default, TiDB prints an alarm log when TiDB memory usage exceeds 70% of its total memory and any of the [alarm conditions](/configure-memory-usage.md#tidb-server-alarm-on-excessive-memory-usage) is met.

Author: ```suggestion
- This variable sets the memory usage ratio that triggers the tidb-server memory alarm. By default, TiDB prints an alarm log when TiDB memory usage exceeds 70% of its total memory and any of the [alarm conditions](/configure-memory-usage.md#trigger-the-alarm-of-excessive-memory-usage) is met.
```

- When this variable is configured to `0` or `1`, it means the memory threshold alarm feature is disabled.

<CustomContent platform="tidb">

- When this variable is configured to a value greater than `0` and less than `1`, it means that the memory threshold alarm feature is enabled.

>     - If the system variable [`tidb_server_memory_limit`](#tidb_server_memory_limit-new-in-v640) is `0` and the [`server-memory-quota`](/tidb-configuration-file.md#server-memory-quota-introduced-from-v409-version) configuration item is not set, the memory alarm threshold is `tidb_memory-usage-alarm-ratio * system memory size`.

Author: ```suggestion
    - If the system variable [`tidb_server_memory_limit`](#tidb_server_memory_limit-new-in-v640) is `0` and the [`server-memory-quota`](/tidb-configuration-file.md#server-memory-quota-new-in-v409) configuration item is not set, the memory alarm threshold is `tidb_memory-usage-alarm-ratio * system memory size`.
```
